Bitbucket.Authentication/Src/Authentication.cs

```diff
@@ -411,11 +411,12 @@ public async Task<Credential> InteractiveLogon(TargetUri targetUri)
         {
             Credential credentials = (Credential)result.Token;
 
-            var realUsername = GetRealUsername(result.RemoteUsername, username);
 
-            if (!targetUri.ContainsUserInfo)
+            // No user info in Uri, or it's a basic login so we need to personalize the credentials.
+            if (!targetUri.ContainsUserInfo || result.Token.Type == TokenType.Personal)
```

Comment:
Tokens set to "Personal" don't have a valid username. We can see that this was a known behavior/had a similar behavior at one point as referenced by the comment in the method summary:

```
/// Bitbucket always wants the username as well as the password/token so if the username
/// isn't explicit in the remote URL then we need to ensure the credentials are stored with a
/// real username rather than 'Personal Access Token' etc
```